src/components/home/AllCollections.js

```diff
@@ -0,0 +1,292 @@
+          (collection) =>
+            collection.name.toLowerCase().includes(query) ||
+            members
+              ?.find(({ id }) => collection.creator === id)
+              ?.name.toLowerCase()
+              .includes(query),
+        ),
+      );
+    }
+  };
+
+  const renderResults = () => {
+    if (!searchResults) {
+      return null;
+    }
+    return (
+      <>
+        <Typography variant="h3" className={classes.typographyMargin}>
+          {t('Search Results')}
+        </Typography>
+        {searchResults.size > 0 ? (
+          <CollectionsGrid collections={searchResults} />
+        ) : (
+          <Typography variant="body1" className={classes.typographyMargin}>
+            {t('No results found.')}
+          </Typography>
+        )}
+      </>
+    );
+  };
+
+  const GOTO_LIST = [
+    '/allCollections',
+    '/preSchool',
+    '/grade1to8',
+    '/highSchool',
+    '/college',
+  ];
+
+  const [sideBarStatus, setSideBarStatus] = React.useState(true);
+
+  const closeSideBar = () => {
+    setSideBarStatus(false);
+  };
+  const OpenSideBar = () => {
+    setSideBarStatus(true);
+  };
+
+  return (
+    <div className={classes.root}>
+      <CssBaseline />
+      <AppBar position="fixed" className={classes.appBar}>
+        <Header />
+      </AppBar>
+      <Drawer
+        className={classes.drawer}
+        variant="persistent"
+        open={sideBarStatus}
+        anchor="left"
+        classes={{
+          paper: classes.drawerPaper,
+        }}
+      >
+        <div className={classes.toolbar} />
+        <div className={classes.drawerHeader}>
+          <IconButton onClick={closeSideBar}>
+            {theme.direction === 'ltr' ? (
+              <ChevronLeftIcon />
+            ) : (
+              <ChevronRightIcon />
+            )}
+          </IconButton>
+        </div>
+        <Divider />
+        <List className={classes.list}>
+          {['All', 'Pre-School', 'Grade 1-8', 'High School', 'College'].map(
+            (text, index) => (
+              <Link href={GOTO_LIST[index]} className={classes.link}>
+                <ListItem button key={text}>
+                  <ListItemIcon>
+                    <BookmarkIcon />
+                  </ListItemIcon>
+                  <ListItemText primary={text} />
+                </ListItem>
+              </Link>
+            ),
+          )}
+        </List>
+        <Divider />
+        <List className={classes.list}>
+          {['Test Prep', 'Post Grad'].map((text) => (
+            <ListItem button key={text} disabled>
+              <ListItemIcon>
+                <BookmarkIcon />
+              </ListItemIcon>
+              <ListItemText primary={text} />
+            </ListItem>
+          ))}
+        </List>
+        <Divider />
+        <a href="https://graasp.eu">
+          <Button variant="contained" className={classes.button}>
+            Create Your Own
+          </Button>
+        </a>
+      </Drawer>
```

Comment:
I think it is not a bad idea to loop on some array. But maybe let's create a new component `MainMenu` to modularize a bit this code.. 

And instead of `['All', 'Pre-School', 'Grade 1-8', 'High School', 'College']` You can loop over 
```
Object.entries({All: 'all-route', "Pre-School":'preschool-route"...}).map([text, link]) 
```
(more or less for the syntax)